plutus-core/untyped-plutus-core/test/Transform/Simplify.hs

```diff
@@ -31,6 +31,26 @@ interveningLambda = runQuote $ do
         arg = mkConstant @Integer () 1
     pure $ Force () $ Apply () lam arg
 
+caseOfCase1 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
+caseOfCase1 = runQuote $ do
+    b <- freshName "b"
+    let ite = Force () (Builtin () PLC.IfThenElse)
+        true = Constr () 0 []
+        false = Constr () 1 []
+        alts = [mkConstant @Integer () 1, mkConstant @Integer () 2]
+    pure $ Case () (mkApplication ite [(Var () b, ()), (true, ()), (false, ())]) alts
```

Comment:
And we should test `nil` & `cons` too, plus maybe some other stuff.

Reply:
Sure, but it's better to test different cases than only 0-ary one (in general tests should assume at little as possible about the concrete thing that they're supposed to test).

Reply:
Added `caseOfCase3`.